src/components/SegmentedTopBar/Indicator.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect } from "react";
+import { StyleSheet } from "react-native";
+
+import PropTypes from "prop-types";
+import Animated, {
+  cancelAnimation,
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+} from "react-native-reanimated";
+import { moderateScale } from "react-native-size-matters";
+
+import { theme } from "@theme";
+
+const springConfig = {
+  mass: 1,
+  overshootClamping: true,
+  restSpeedThreshold: 0.5,
+  restDisplacementThreshold: 0.5,
+};
+
+const spacing = moderateScale(6);
+
+export const Indicator = ({ measure }) => {
+  const { x, width, height } = measure;
+
+  const translateX = useSharedValue(0);
+  const animatedWidth = useSharedValue(0);
+
+  useEffect(() => {
+    cancelAnimation(translateX);
+    cancelAnimation(animatedWidth);
+
+    translateX.value = withSpring(x + spacing / 2, springConfig);
+    animatedWidth.value = withSpring(width - spacing, springConfig);
+  }, [animatedWidth, translateX, width, x]);
+
+  const tabTranslateAnimatedStyles = useAnimatedStyle(() => ({
+    transform: [{ translateX: translateX.value }],
+    width: animatedWidth.value,
+  }));
+
+  return (
+    <Animated.View
+      style={[
+        styles.movingSegmentStyle,
+        { height: height - spacing },
+        tabTranslateAnimatedStyles,
+      ]}
+    />
+  );
+};
+
+const styles = StyleSheet.create({
+  movingSegmentStyle: {
+    position: "absolute",
+    top: spacing / 2,
+    borderRadius: moderateScale(6),
+    backgroundColor: theme.colors.background.primary,
+    elevation: 5,
+    shadowColor: theme.colors.background.base,
```

Comment:
@chiragbhaiji Shadow looks too much blue in iOS.

<img width="576" alt="Screenshot 2023-03-25 at 9 35 11 AM" src="https://user-images.githubusercontent.com/93119254/227693631-dc33b5f8-811d-4dd3-a73f-c41973bcc5f8.png">